internal/credentials/spiffe/spiffe.go

```diff
@@ -40,27 +44,38 @@ type partialParsedSPIFFEBundleMap struct {
 // If duplicate keys are encountered in the JSON parsing, Go's default unmarshal
 // behavior occurs which causes the last processed entry to be the entry in the
 // parsed map.
-func LoadSPIFFEBundleMap(filePath string) (map[string]*spiffebundle.Bundle, error) {
+func LoadSPIFFEBundleMap(filePath string) (BundleMap, error) {
 	bundleMapRaw, err := os.ReadFile(filePath)
 	if err != nil {
 		return nil, err
 	}
+	return BundleMapFromBytes(bundleMapRaw)
+}
+
+// BundleMapFromBytes parses bytes into a SPIFFE Bundle Map. See the
+// SPIFFE Bundle Map spec for more detail -
+// https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Trust_Domain_and_Bundle.md#4-spiffe-bundle-format
+// If duplicate keys are encountered in the JSON parsing, Go's default unmarshal
+// behavior occurs which causes the last processed entry to be the entry in the
+// parsed map.
+func BundleMapFromBytes(bundleMapBytes []byte) (BundleMap, error) {
 	var result partialParsedSPIFFEBundleMap
-	if err := json.Unmarshal(bundleMapRaw, &result); err != nil {
+	err := json.Unmarshal(bundleMapBytes, &result)
+	if err != nil {
```

Comment:
Nit: why is this change of splitting one line into two required?

Reply:
Good catch, this was something I fixed earlier but I think some auto-merging brought it back